analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LockedFieldShouldBeReadonly.cs

```diff
@@ -214,3 +214,21 @@ namespace ANamespace
 {
     class AClass { }
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/7058
+class Repro_7058
+{
+    void Test(int[] a)
+    {
+        var local = new object();
+        Action action1 = () =>
+        {
+            lock (local) { } // Noncompliant, FP: local is in the captured scope
+        };
+        Action action2 = () =>
+        {
+            var localCaptured = local;
+        };
+        lock (local) { }     // Noncompliant, locking captured variable in its declaration scope
+    }
```

Comment:
```suggestion
    }

    void LockOutsideCapture()
    {
        var local = new object();
        Action action = () =>
        {
            var localCaptured = local;
        };
        lock (local) { }     // Noncompliant, locking captured variable in its declaration scope
    }
```

Reply:
Changed.
Splitting into two methods and having specific names is, however, in this scenario and in my opinion, of very little help.
The non-compliance comment, which is right next to the primary position by importance, explains well the nature of the non-compliance. Repeating it 3-4 lines away from that location, in the name of the method, is not useful.
Moreover, it should be considered nitpicking and not blocking for the PR.